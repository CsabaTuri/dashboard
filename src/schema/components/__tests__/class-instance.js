import React from "react"
import { Map } from "immutable"
import { shallow } from "enzyme"
import { shallowToJson } from "enzyme-to-json"

import ClassInstance from "../class-instance"
import { reduceSchemaTree } from "../../"

describe("ClassInstance Component", () => {
  let comp, store, client

  beforeEach(() => {
    client = { query: jest.fn(() => Promise.resolve({ name: "fake-instance" })) }
    store = createImmutableTestStore({ schema: reduceSchemaTree })()

    comp = shallow(
      <ClassInstance
        store={store}
        client={client}
        path={["a-db"]}
        clazz={Map.of("name", "fake-class")} />
    ).dive()
  })

  it("renders instance form", () => {
    expect(shallowToJson(comp)).toMatchSnapshot()
  })

  it("create a new instance", () => {
    comp.find("ReplEditor").simulate("change", "{ 'name': 'Bob' }")
    expect(shallowToJson(comp)).toMatchSnapshot()

    return store.dispatch(comp.find("Connect(SchemaForm)").props().onSubmit()).then(() => {
      comp.update()
      expect(shallowToJson(comp)).toMatchSnapshot()
    })
  })
})


