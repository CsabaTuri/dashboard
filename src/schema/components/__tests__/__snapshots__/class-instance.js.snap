exports[`ClassInstance Component create a new instance 1`] = `
<div>
  <Connect(SchemaForm)
    buttonText="Create Instance"
    onSubmit={[Function]}
    title="Create an instance of fake-class">
    <ReplEditor
      mode="editor form-field text-area"
      name="class-data-editor"
      onChange={[Function]}
      value="{ \'name\': \'Bob\' }" />
    <p
      className="ms-TextField-description">
      The contents of this field will be evaluated with the context of a repl and placed in the \"data\" field of the new instance.
    </p>
  </Connect(SchemaForm)>
  <Connect(InstanceInfo)
    instance={null} />
</div>
`;

exports[`ClassInstance Component create a new instance 2`] = `
<div>
  <Connect(SchemaForm)
    buttonText="Create Instance"
    onSubmit={[Function]}
    title="Create an instance of fake-class">
    <ReplEditor
      mode="editor form-field text-area"
      name="class-data-editor"
      onChange={[Function]}
      value="{ \'name\': \'Bob\' }" />
    <p
      className="ms-TextField-description">
      The contents of this field will be evaluated with the context of a repl and placed in the \"data\" field of the new instance.
    </p>
  </Connect(SchemaForm)>
  <Connect(InstanceInfo)
    instance={
      Object {
        "name": "fake-instance",
      }
    } />
</div>
`;

exports[`ClassInstance Component renders instance form 1`] = `
<div>
  <Connect(SchemaForm)
    buttonText="Create Instance"
    onSubmit={[Function]}
    title="Create an instance of fake-class">
    <ReplEditor
      mode="editor form-field text-area"
      name="class-data-editor"
      onChange={[Function]}
      value="{}" />
    <p
      className="ms-TextField-description">
      The contents of this field will be evaluated with the context of a repl and placed in the \"data\" field of the new instance.
    </p>
  </Connect(SchemaForm)>
  <Connect(InstanceInfo)
    instance={null} />
</div>
`;
